### ISSUE TITLE
The action add is not defined in controller CoreIngredientsController

### ISSUE THREAD

**Issue by zasmth** [nazgul26/PHPRecipebook#7]
This is a great app, but I'm getting a number of errors. I've tried on MySQL and PostgreSQL. I've tried versions 4.09, 5.0.0, 5.0.1, and 5.0.2. I've tried both with and without git. I really love this app and want to get it working, but no matter what I try, I'm seeing errors like these:

Error: The action add is not defined in controller CoreIngredientsController
Error: Create CoreIngredientsController::add() in file: recipe/Controller/CoreIngredientsController.php.

``` <?php
class CoreIngredientsController extends AppController {
    public function add() {
    }
}
```

From the ingredients page, when I click add unit from the dropdown I get the error:
Error: The action add is not defined in controller UnitsController
Error: Create UnitsController::add() in file: recipe/Controller/UnitsController.php.

```
<?php
class UnitsController extends AppController {


    public function add() {

    }

}
```

But, when I click list units and then click add unit from the list page, it works. Something strange is happening...

Also, I had major trouble importing the core_ingredients.sql file to the database. In the file, groupNumber is used, but in the database it is group_id. In the file, weights are called recipe_core_weights, but the database column is just core_weights. I got the file to import by changing the file to match the column titles, but why did I have to do that?

I wasn't able to install 5.0.2 because some ___d() error. I forget what it was now, since I've been focusing on 5.0.1 since I managed to get that the furthest.

I'm not one to post online for help, I created an account just for this, but I've been at it for days and think I've exhausted google. Thank you for making this great app, and thank you for your help with my problems!

**Comment by nazgul26** [comment]
Glad you like what you have seen so far.

So for your issues:
1. CoreIngredients add() method. I couldn't find the link that called this.  What page were you on and called it?  It should call 'edit' action. Not sure what one I missed.
2. UnitsController add() method.  See the issue and fixed. Will call 'edit' and open in dialog like the rest.
3. CoreIngredients import script.  Fixed.  I had plans to have the table use the CakePHP naming convention but that did not work out.  I have put it back the way it has to be which is with groupNumber as the field.
4. In the setup 5.0.2 wizard fixed the called d() which should have called __d().  

So give me some more details on #1 and anything else you find and then i will package up for a release.

Todd

**Comment by zasmth** [comment]
Awesome, thanks for the fast reply! Ok, I've tried to be thorough, hopefully this will be helpful.

1 On the Core Ingredients page, the Add Core Ingredient link (/core_ingredients/add)

Other things I noticed:
2 If I have a shopping list with a recipe on it and then I go to the recipe page and delete the recipe, I get sent to a white page as soon as I click delete that says: Notice (8): Undefined index: image [APP/Plugin/Upload/Model/Behavior/UploadBehavior.php, line 1987]

Then on the shopping list page errors are shown:
Notice (8): Undefined index: name [APP/View/ShoppingLists/index.ctp, line 97]
Notice (8): Undefined index: id [APP/View/ShoppingLists/index.ctp, line 98]
Notice (8): Undefined index: serving_size [APP/View/ShoppingLists/index.ctp, line 113]

Other side effects of this (which may only apply when you delete all of your recipes, I only had one recipe and I deleted it, not sure what would happen if I still had recipes left):

Creating a new recipe gives errors: http://i.imgur.com/V8NCkad.png
Error: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'ingredient_id' cannot be null
SQL Query: INSERT INTO `recipe`.`ingredient_mappings` (`recipe_id`, `ingredient_id`, `sort_order`, `quantity`, `unit_id`, `qualifier`, `optional`) VALUES (3, NULL, 0, NULL, 1, '', '0')

The side menu also disappeared because of this. It is missing when logged out also.

3 From the ingredients page, clicking More Actions... > Add Core Ingredient brings up a popup to add an ingredient, but there is not a box for name. Only Group Number, Description, and Short Description are shown. (Related question, what is group number for?)

4 Very minor issue, but on the Core Ingredients page, the More Actions... dropdown shows up even when there are no other actions. Not something that needs to be changed because nothing is actually broken, but I find myself clicking it often to see what the other actions are.

5 In the Add Ingredient popup, Solid/Liquid is followed by a yes/no dropdown. What's the intention for this field?

6 What's the purpose of the store layout textbox?

I also have two feature requests, let me know what you think:
7 A health level for recipes, this could just be 1-Healthy, 2-Neutral, 3-Unhealthy

8 Ability for admin to force or toggle sharing all ingredients/recipes. I realize when creating a new user I can give them my ingredients, but after their account has been created, any ingredients I add are not seen by them. Maybe this won't be necessary when core ingredients are working.

9 Does http://sourceforge.net/p/phprecipebook/donate/ still go to you? I'd like to support the project, but want to make sure that's still the way to do it.

Thanks again!

**Comment by nazgul26** [comment]
#1 fixed and in 5.0.3 release.  The delete issue sounds a bit bigger. Will work through the rest and spin out the bigger items to separate tickets.

For the feature requests;
- Health level:  Sorry but not a big fan of the idea. I could help you through if you want to add to your local copy.  In this re-write I actually dropped a couple features.  So I would probably go back to finishing that up before I did this one.
- Ingredient Copy: This one I have gone in many different directions of the years.  There are 2 different use cases for the RecipeBook. The first is single user and then you want to all share recipes.  Then there is a multi-user setup and those will want to tweak their own after getting a boiler plate setup.  So I tried to make it work mostly out the box for both.  The only thing I can think of next is to try an admin sync/copy option that would send out some ingredients to all users.  Not sure how much value that would be.
- The donate link might still work, but to honest I don't trust SourceForge anymore.  Ever since I got an adware infested of WinSCP from them it has not been cool.  So for now don't worry about donations, I just do it cause I like to.

**Comment by nazgul26** [comment]
Opened new issue to track the delete of a recipe while in shopping list. Closing this one.
